mantidproject/mantid
# Engineering Diffraction GUI now supports fitting multiple runs

Multi-run fitting from the ED GUI was wired up properly, and a large amount of refactoring was done, including the removal of a **lot** of dead code from the view and the presenter (this was mainly code related to making sure files exist, which is now taken care of in the model). Users should now be able to load several workspaces and apply Engg fitting algorithms to them all in one go.

**To test:**

Some sample data is [here](https://github.com/mantidproject/mantid/files/1460806/ENGINX_focused_runs.zip).

Load a few of the other runs by clicking the first **Browse** button, selecting the focused runs you want and then hitting **Load**.

Press **Fit all**. A table workspaces with fit parameters for all runs should appear in the workspaces area, and a matrix workspaces with all the fitted peaks should also appear.

Check that fitting information was saved (as `.csv`). On Windows this will be in `C:\EnginX_Mantid\User`, on Linux `~/EnginX_Mantid/User`.

When you select a different run from the list widget, the fit results should be plotted over it - make sure this works nicely.

Fixes #21247 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/c2b9dffae50f214d0fbb912517e3cbd5f5138c3d)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
